internal/provider/telemetry_resource_test.go

```diff
@@ -931,12 +921,187 @@ resource "modtm_telemetry" "test" {
+		endpointAssignment = fmt.Sprintf("endpoint = \"%s\"", endpointAssignment)
+	}
+	enabledAssignment := ""
+	if !enabled {
+		enabledAssignment = "enabled = false"
+	}
+	sb := strings.Builder{}
+	for k, v := range tags {
+		sb.WriteString(fmt.Sprintf("%s = \"%s\"", k, v))
+		sb.WriteString("\n")
+	}
+	r := fmt.Sprintf(`
+provider "modtm" {
   %s
+  %s
+}
+
+resource "modtm_telemetry" "test" {
+  tags = {
+   %s
+  }
+	module_path = "%s"
 }
-`, endpointAssignment, enabledAssignment, sb.String(), nonceAssignment)
+`, endpointAssignment, enabledAssignment, sb.String(), modulePath)
 	return r
 }
 
-func intPtr(i int) *int {
-	return &i
+func TestModulePathToKey_Success(t *testing.T) {
+	t.Setenv("TF_DATA_DIR", ".terraform")
+	defer os.Unsetenv("TF_DATA_DIR")
+
+	tests := []struct {
+		name       string
+		modulePath string
+		expected   string
+	}{
+		{"Valid module path", ".terraform/modules/abc123", "abc123"},
+		{"Nested module path", ".terraform/modules/nested/abc123", "nested"},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			key, err := modulePathToKey(tt.modulePath)
+			assert.NoError(t, err)
+			assert.Equal(t, tt.expected, key)
+		})
+	}
+}
+
+func TestModulePathToKey_Failure(t *testing.T) {
+	tests := []struct {
+		name       string
+		modulePath string
+	}{
+		{"Non-matching module path", "/path/to/nonexistent"},
+		{"Empty module path", ""},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := modulePathToKey(tt.modulePath)
+			assert.Error(t, err)
+		})
+	}
+}
+
+func TestModulePathToKey_WithCustomTFDataDir(t *testing.T) {
+	customTFDataDir := "/custom/terraform/data"
+	t.Setenv("TF_DATA_DIR", customTFDataDir)
+	defer os.Unsetenv("TF_DATA_DIR")
+
+	modulePath := "/custom/terraform/data/modules/xyz789"
+	expectedKey := "xyz789"
+
+	key, err := modulePathToKey(modulePath)
+	assert.NoError(t, err)
+	assert.Equal(t, expectedKey, key)
+}
+
+func TestModulePathToKey_InvalidInput(t *testing.T) {
+	_, err := modulePathToKey("")
+	assert.Error(t, err)
+}
+
+// createModulesJson creates a modules.json file with a reference to a standard module (kv) and
+// reference to a child module of the key vault module (keys) in the root module.
+func createModulesJson() error {
+	if err := os.MkdirAll(".terraform/modules", 0755); err != nil {
+		return err
+	}
+	f, err := os.Create(".terraform/modules/modules.json")
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	_, err = f.WriteString(`{
+  "Modules": [
+    {
+      "Key": "",
+      "Source": "",
+      "Dir": "."
+    },
+    {
+      "Key": "keys",
+      "Source": "registry.terraform.io/Azure/avm-res-keyvault-vault/azurerm//modules/key",
+      "Version": "0.6.1",
+      "Dir": ".terraform/modules/keys/modules/key"
+    },
+    {
+      "Key": "kv",
+      "Source": "registry.terraform.io/Azure/avm-res-keyvault-vault/azurerm",
+      "Version": "0.6.1",
+      "Dir": ".terraform/modules/kv"
+    },
+    {
+      "Key": "kv.keys",
```

Comment:
Should we add test case for this item? Could our current implementation handle this module correctly?